### PR TITLE
docs: remove duplicated package versions description

### LINKED ISSUE #7341: Docs: Two pages describe package versions
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

Both https://typescript-eslint.io/users/dependency-versions and https://typescript-eslint.io/users/versioning exist say what versions we support right now. I haven't looked deeply, but I'm guessing there was a merge conflict that resulted in the pages repeating each other.

### Affected URL(s)

https://typescript-eslint.io/users/dependency-versions
https://typescript-eslint.io/users/versioning

### PR DESCRIPTION
**Pull request by auvred**
## PR Checklist

- [x] Addresses an existing open issue: fixes #7341
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Reverting hunk from this merge commit https://github.com/typescript-eslint/typescript-eslint/commit/c23365292631b1c3db7c439c9d962234084f66b7, according to https://github.com/typescript-eslint/typescript-eslint/issues/7341#issuecomment-1654584927

### CODE REVIEW

File: docs/users/Versioning.mdx [hunk 1]
```diff
@@ -6,74 +6,6 @@ title: Versioning
...
-
-With each new TypeScript release we file an issue to track the changes in the new version. The issue should always be pinned, and you can also [find the issues by searching for issues tagged with "New TypeScript Version"](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22New+TypeScript+Version%22+sort%3Acreated-desc). If the issue is open, we do not have official support yet - please be patient.
-
-In terms of what versions we support:
-
-- We do not support the `beta` releases.
-- We _generally_ do not officially support the `rc` releases.
-- We endeavor to support the latest stable TypeScript versions as soon as possible after the release.
-
-Generally we will begin working on supporting the next release when the `rc` version is released.
```
Comment (JoshuaKGoldberg) [inline docs/users/Versioning.mdx:67]: Ah, there's information in this page that's not in the other one. Could you move them over please?

Just skimming:

* Mentioning the DefinitelyTyped support window
* Not accepting issues against unsupported versions
* Not supporting beta or rc versions, but trying to support stables soon after release
* Having a _Version Warning Logs_ heading that can be linked to

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Thanks for starting on this! I think the general direction is good, but because of the merge shenanigans there's additional info on this page we'd want to preserve.

File: docs/users/Versioning.mdx (hunk 1 above)
Comment (auvred) [reply docs/users/Versioning.mdx:67]: Oh, sorry, I missed this! I'll bring it back

---
One small question:

> Having a Version Warning Logs heading that can be linked to

This info is present [here](https://typescript-eslint.io/users/dependency-versions/#typescript)

https://github.com/typescript-eslint/typescript-eslint/blob/54ef459e0250a7d0d3ddad4ce1a458d003126383/docs/users/Dependency_Versions.mdx#L29-L31

Should I place it on `Versioning` page too?

Comment (auvred) [reply docs/users/Versioning.mdx:67]: Or I need to add this heading to `Dependency_Versions` and add link to it from `Versioning`?

Comment (JoshuaKGoldberg) [reply docs/users/Versioning.mdx:67]: Yeah, the info is there but a heading isn't. The heading is nice for discoverability and so folks can link directly to it.

Comment (JoshuaKGoldberg) [reply docs/users/Versioning.mdx:67]: To be clear, I'm saying it's fine to delete this all out of `Versioning.mdx`, just put the `#### Version Warning Logs` above the same existing info in `Dependency_Versions.mdx`. That way we stick with having one place for the info.

**Comment by auvred** [comment]
I guess I added all missed stuff :)

File: docs/users/Versioning.mdx [hunk 2]
```diff
@@ -6,74 +6,6 @@ title: Versioning
 We follow [semantic versioning (semver)](https://semver.org).
 This page exists to help set guidelines around when what we consider to fall within each of the semver categories.
```
Comment (JoshuaKGoldberg) [inline docs/users/Versioning.mdx:7]: Ooh just found an existing typo 😄 

```suggestion
This page exists to help set guidelines around what we consider to fall within each of the semver categories.
```

File: docs/users/Versioning.mdx [hunk 3]
```diff
@@ -4,75 +4,7 @@ title: Versioning
 ---
 
 We follow [semantic versioning (semver)](https://semver.org).
-This page exists to help set guidelines around when what we consider to fall within each of the semver categories.
-
-All of the packages in this project are published with the same version number to make it easier to coordinate both releases and installations.
```
Comment (JoshuaKGoldberg) [inline docs/users/Versioning.mdx:9]: I don't see this sentence anywhere else in the docs, do you?

File: docs/users/Versioning.mdx (hunk 2 above)
Comment (auvred) [reply docs/users/Versioning.mdx]: Nice! Commited it

File: docs/users/Versioning.mdx [hunk 4]
```diff
@@ -4,75 +4,7 @@ title: Versioning
...
-
-We publish a canary release on every successful merge to `main`, so **you never need to wait for a new stable version to make use of any updates**.
```
Comment (JoshuaKGoldberg) [inline docs/users/Versioning.mdx:11]: The idea of a canary release is mentioned in the _Releases_ page, but the emphasis on _never need to wait for a new stable version_ I don't see anywhere else. Do you?

File: docs/users/Versioning.mdx [hunk 5]
```diff
@@ -4,75 +4,7 @@ title: Versioning
...
-  <img
-    src="https://img.shields.io/npm/v/@typescript-eslint/parser/canary.svg?style=flat-square"
-    alt="NPM Version"
-  />
-</a>
-
-:::note
-The only exception to the automated publishes described above is when we are in the final phases of creating the next major version of the libraries - e.g. going from `1.x.x` to `2.x.x`.
-During these periods, we manually publish `canary` releases until we are happy with the release and promote it to `latest`.
-:::
```
Comment (JoshuaKGoldberg) [inline docs/users/Versioning.mdx:36]: Is this note anywhere else? I don't see it.

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Very close! Just a bit more stuff that I don't see in the new form. I think we'll be good to go once that's added back in (or I'm told I just did a silly and missed them?). 🚀

**Comment by auvred** [comment]
Excuse me for my inattention, documentation is not my strongest point 😓

**Review by JoshuaKGoldberg — APPROVED** [review]
Wonderful, thanks! 🙌

**Pull request merged**